lib/wlib/memory/Memory.cpp

```diff
@@ -144,21 +241,41 @@ extern "C" Allocator *memory_get_allocator(size_t size) {
 /**
  * Allocates a memory block of the requested size. The blocks are created from
  * the fixed block allocators
+ *
  * @param size the client's requested size of the block
  * @return a pointer to the memory block
  */
-extern "C" void *memory_alloc(size_t size) {
+void *memory_alloc(uint32_t size) {
     // Allocate a raw memory block
     Allocator *allocator = memory_get_allocator(size);
+
+    // if not enough sizes available. This will only be the
+    // case for pools
+    if (allocator == nullptr)
+        return nullptr;
+
     void *blockMemoryPtr = allocator->Allocate();
 
+    // if not enough quantity of that particular size available
+    if (blockMemoryPtr == nullptr) {
+        // check if higher sizes have some blocks available and
+        // give a block from them if available
+        uint16_t currBlockSize = allocator->GetBlockSize();
+        while (currBlockSize < _allocators[MAX_ALLOCATORS - 1]->GetBlockSize())
+            return memory_alloc(currBlockSize + 1);

```

Comment:
Function will return upon first loop execution, so couldn't this just be an `if` statement?

Reply:
That is true it can be let me fix that